src/Cake.NuGet/NuGetContentResolver.cs

```diff
@@ -72,6 +76,11 @@ private IReadOnlyCollection<IFile> GetAddinAssemblies(DirectoryPath path, Packag
                 // Get relative path.
                 var relative = path.GetRelativePath(assembly.Path);
                 var framework = ParseFromDirectoryPath(current, relative.GetDirectory());
+                if (framework == null)
+                {
+                    _log.Warning("Unable to parse compatible framework from path {0}. Setting any framework.", assembly.Path.MakeAbsolute(_environment).FullPath);
```

Comment:
It's actually the other way round. For each assembly, we will take the directory and try to parse the TFM and map that with the assemblies in that directory. After looping through all assemblies, we will find the closest matching our Target Framework (.NET Framework, Version=4.6 or .NET Standard, Version=1.6) and return all files mapped to that.

Now when looking over code once more, I found a smarter way to log warning. Now we log for each assembly found in root. Instead, we should only log if assemblies mapped to `AnyFramework` is actually returned. Let me modify this PR slightly :smile: